packages/core/docs/general/context.md

```diff
@@ -12,137 +18,159 @@ $vsf {
   ...
 }
 ```
+
 - `$vsf` - general key that keeps vue storefront context
 - `$ct` - integration key
-- `api` - field that always keep api function for given integration
+- `api` - field that always keep api functions for given integration
 - `client` - field that always keep api client/connection for given integration
 - `config` - field that always keep configuration for given integration
 - others - depending on the needs you can put into the context any field you want (under the corresponding key)
 
-## Usage in api client functions
-IMPORTANT: The api-client context is not the one within the entire app, it gives you only access to configuration and connection, it doesn't know anything about the application logic
+## Api-client creation
 
-```js
-const createCart = async ({ config, client }, cartDraft, customQueryFn?) => {
-  // config - access to current config
-  // client - access to current api connection / client
+Since the api-client can be used as separated unity, and it's responsible for communication with the given integration, it is also related to the context API.
+
+Each api function has context in the first argument that contains information about `client` and the `config`.
 
-  const { locale, acceptLanguage, currency } = config;
+In terms of creation of context api, we need to pass each api-function to the factory params of `apiClientFactory` along with `tag` for the integration and `onSetup` function (if it's needed):
+
+```js
+import { apiClientFactory } from '@vue-storefront/core';
+import getProduct from './api/getProduct';
+import getCategory from './api/getCategory';
 
-  const request = await client.mutate({
-    mutation: gql`${query}`,
-    variables: { locale, acceptLanguage, currency },
-    fetchPolicy: 'no-cache'
+const onSetup = (settings) => {
+  const client = new ApolloClient({
+    link: apolloLink,
+    cache: new InMemoryCache(),
   });
 
-  return request;
+  return {
+    config,
+    client
+  };
 };
-```
 
-## Usage in components
+const { createApiClient } = apiClientFactory({
+  tag: 'ct',
+  onSetup,
+  api: {
+    getProduct,
+    getCategory,
+  }
+});
 
-```js
-import { useVSFContext } from '@vue-storefront/core'
+export { createApiClient }
+```
 
-setup () {
-  const { $ct } = useVSFContext();
+The `tag` in the configuration is sort of identifier what will be used as a context key.
 
-  // $ct.api
-  // $ct.client
-  // $ct.config
-}
-```
+Inside of `onSetup` function you can put a creation of the connection or anything else that's needed for preparing an api connection - remember, onSetup must always return the `client` and `config`.
+
+## Api client usage
```

Comment:
missing info (tip) like: 'If you're using our default setup and Nuxt you can omit this part" or similar